File: semana 1/index_v2.php
<?php

$path_in  = $argv[1];
$path_out = $argv[2];
$fr       = fopen($path_in,  'r');

$pre_line = array(
    'n00,n01,A'. \PHP_EOL,
    'n10,n11,B'. \PHP_EOL,
);
$csv_out = '';

/* read & pre-write */
while(($csv_in_line = fgetcsv($fr)) !== FALSE)
{
    /* real value? */
    if(is_null(($true = array_search('TRUE', $csv_in_line))) || !is_numeric($csv_in_line[0]))
    {
        continue;
    }
    // A/B x/y
    $ab    = (int)($true > 2);
    $match = 'n'. $ab . (($true - 1) % 2) .'';
    
    if(!strpos($csv_out, $match))
    {
        $csv_out .= $pre_line[$ab];
    }
    // replace match -> code
    $csv_out = substr_replace($csv_out, $csv_in_line[0], strpos($csv_out, $match), strlen($match));

}

// clean no replace
$csv_out = str_replace(array('n00','n01','n10','n11'), array('0','0','0','0'), $csv_out);

fclose($fr);

file_put_contents($path_out, $csv_out);

File: semana 1/index_v2.py
import sys
import csv
import string

path_in  = sys.argv[1]
path_out = sys.argv[2]

fr           = open(path_in, 'rb')
lines_in_csv = csv.reader(fr)
out_csv      = ''

pre_match = ['n00,n01,A\n','n10,n11,B\n']

for index,line in enumerate(lines_in_csv) :


    if not line[0].isdigit() or not 'TRUE' in line :
        continue

    true_pos = line.index('TRUE')
    ab       = int(true_pos > 2)
    match    = 'n' + str(ab) + str((true_pos - 1) % 2) + ''

    if not match in out_csv :
        out_csv += pre_match[ab]

    out_csv = out_csv.replace(match, line[0], 1)

out_csv = out_csv.replace('n00','0')
out_csv = out_csv.replace('n01','0')
out_csv = out_csv.replace('n10','0')
out_csv = out_csv.replace('n11','0')

fw = open(path_out,'wb')
fw.write(out_csv)
fw.close()
